src/c2d.jl

```diff
@@ -42,31 +41,31 @@ end
   Cd    = C
   Dd    = D + C*M2
   x0map = [speye(nx) -M2]
-  StateSpace{Siso{T}}(Ad, Bd, Cd, Dd, Ts), x0map
+  ss(Ad, Bd, Cd, Dd, Ts), x0map
 end
 @compat (m::FOH)(s::LtiSystem, Ts::Real) = (m::FOH)(ss(s), Ts)[1]
 
 
 immutable GeneralizedBilinear{T} <: Method
   α::T
   @compat function (::Type{GeneralizedBilinear}){T}(α::T = 0.5)
-    @assert α ≧ 0. && α ≦ 1. "GeneralizedBilinear: α must be between 0 and 1"
+    @assert α ≥ 0. && α ≤ 1. "GeneralizedBilinear: α must be between 0 and 1"
     new{T}(α)
   end
 end
 
-@compat function (m::GeneralizedBilinear){T}(s::StateSpace{Siso{T}}, Ts::Real)
+@compat function (m::GeneralizedBilinear){T}(s::StateSpace{Val{T}}, Ts::Real)
   A, B, C, D  = s.A, s.B, s.C, s.D
   ny, nu      = size(s)
-  nx          = numstates(s)
+  nx          = size(A,1)
   α           = m.α
-  ima   = speye(nx) - α*Ts*A
-  Ad    = ima\(speye(nx) + (1.0-α)*Ts*A)
+  ima   = I - α*Ts*A
+  Ad    = ima\(I + (1.0-α)*Ts*A)
   Bd    = ima\(Ts*B)
   Cd    = (ima.'\C.').'
   Dd    = D + α*(C*Bd)
   x0map = [speye(nx) zeros(nx, nu)]
-  StateSpace{Siso{T}}(Ad, Bd, Cd, Dd, Ts), x0map
+  ss(Ad, Bd, Cd, Dd, Ts), x0map
```

Comment:
Same here. Maybe we can have some helper function for these conversions to avoid the enforcing inner constructor.